specs/_features/eip7547/fork-choice.md

```diff
@@ -23,13 +24,67 @@ This is the modification of the fork choice accompanying EIP-7547.
 ### New `is_inclusion_list_available`
 
 ```python
-def is_inclusion_list_available(self: ExecutionEngine, new_payload_request: NewPayloadRequest) -> bool:
+def is_inclusion_list_available(block_root: Root) -> bool:
     """
     Return ``True`` if and only if the payload has a corresponding inclusion list.
     """
     ...
 ```
 
+### Modified `filter_block_tree`
+
+Add filter for if the inclusion list is available.
+
+```python
+def filter_block_tree(store: Store, block_root: Root, blocks: Dict[Root, BeaconBlock]) -> bool:
+    block = store.blocks[block_root]
+    # [Modified in EIP-7547] Check that IL is available when considering heads.
+    children = [
+        root for (root, block) in blocks.items()
+        if block.parent_root == block_root
+        and is_inclusion_list_available(root)
```

Comment:
This doesn't work since you may have grand children that are valid

Reply:
Could you make this work by doing the following: 

- as you add something to forkchoice,  include an "IL available bool", store this bool along with the block
- if the bool is true, iterate back to the justified root flipping all to true
- `is_inclusion_list_available` here just checks this bool is true

Effectively tracking the latest seen block with an available IL which I think is what we want

Reply:
This doesn't really work because we may have to reorg back to a head for witch we haven't seen an IL. Without reconstructing it we always rely on external sources

Reply:
> Could you make this work by doing the following:
> 
> * as you add something to forkchoice,  include an "IL available bool", store this bool along with the block
> * if the bool is true, iterate back to the justified root flipping all to true
> * `is_inclusion_list_available` here just checks this bool is true
> 
> Effectively tracking the latest seen block with an available IL which I think is what we want
Not sure if this was after or before the Discord discussion, but just in case and to leave it documented, this doesn't work either since you may be turning for true a node where no one has seen the IL except the child's builder. In this case this node should never be taken for head or the EL should be able to reconstruct the IL.